Adopt smart pointers for updateLayout calls

rwlbuis opened the pull request:
#### fc08a3b096a4d7587bade1e0f7a7c6074a9d8537
<pre>
Adopt smart pointers for updateLayout calls
<a href="https://bugs.webkit.org/show_bug.cgi?id=272738">https://bugs.webkit.org/show_bug.cgi?id=272738</a>

Reviewed by Ryosuke Niwa.

Adopt smart pointers for updateLayout calls.

* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::propertyValue const):
* Source/WebCore/dom/BoundaryPoint.h:
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::firstRectForRange const):
* Source/WebCore/html/HTMLLabelElement.cpp:
(WebCore::HTMLLabelElement::focus):
* Source/WebCore/html/ImageInputType.cpp:
(WebCore::ImageInputType::height const):
(WebCore::ImageInputType::width const):
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.cpp:
(WebCore::MediaControlTextTrackContainerElement::createTextTrackRepresentationImage):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::performSizeToContentAutoSize):
(WebCore::LocalFrameView::setTracksRepaints):
(WebCore::LocalFrameView::trackedRepaintRectsAsText const):
* Source/WebCore/page/PrintContext.cpp:
(WebCore::PrintContext::pageProperty):
* Source/WebCore/page/ios/FrameIOS.mm:
(WebCore::LocalFrame::wordsInCurrentParagraph const):
(WebCore::LocalFrame::preferredHeight const):
(WebCore::LocalFrame::updateLayout const):
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
(WebKit::WebFullScreenManager::willEnterFullScreen):
* Source/WebKit/WebProcess/InjectedBundle/DOM/InjectedBundleRangeHandle.cpp:
(WebKit::InjectedBundleRangeHandle::text const):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::updateAttachmentAttributes):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::requestDocumentEditingContext):
(WebKit::WebPage::focusTextInputContextAndPlaceCaret):

Canonical link: <a href="https://commits.webkit.org/277606@main">https://commits.webkit.org/277606@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1a4315f5c8789e4fab11609520730b49ffbf3511

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48014 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27218 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50924 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50697 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44071 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/50321 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33104 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24710 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/39177 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48597 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/24892 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/41535 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20358 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22371 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/42711 "Found 1 new test failure: fast/css/viewport-height-border.html (failure)") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6061 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/44362 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43145 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52633 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23057 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/19412 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/46612 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24326 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/41682 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45486 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/10613 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25125 "Built successfully") | | | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24048 "Hash 1a4315f5 for PR 27318 does not build (failure)") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/editing/Editor.cpp:
```diff
@@ -3648,7 +3648,8 @@ static inline void collapseCaretWidth(IntRect& rect)
 {
-    range.start.document().updateLayout();
+    Ref document = range.start.document();
+    document->updateLayout();
```
Why not just range.start.protectedDocument()->updateLayout()?

rwlbuis replied:
Fixed.

rniwa commented on Source/WebCore/page/LocalFrameView.cpp:
```diff
@@ -5337,8 +5337,8 @@ void LocalFrameView::setTracksRepaints(bool trackRepaints)
     if (trackRepaints) {
-        if (m_frame->document())
-            m_frame->document()->updateLayout(LayoutOptions::UpdateCompositingLayers);
+        if (auto document = m_frame->protectedDocument())
```
Do: `RefPtr document = m_frame->document()` instead.

rwlbuis replied:
Fixed.

rniwa commented on Source/WebCore/page/LocalFrameView.cpp:
```diff
@@ -5365,7 +5365,7 @@ String LocalFrameView::trackedRepaintRectsAsText() const
     Ref protectedFrame { frame };
 
-    if (auto* document = frame.document())
+    if (auto document = frame.protectedDocument())
```
Ditto.

rwlbuis replied:
Fixed.

rniwa commented on Source/WebCore/page/PrintContext.cpp:
```diff
@@ -364,11 +364,11 @@ String PrintContext::pageProperty(LocalFrame* frame, const char* propertyName, i
     Ref protectedFrame { *frame };
 
-    auto& document = *frame->document();
+    auto document = frame->protectedDocument();
```
Ditto: `RefPtr document = frame->document();`

rwlbuis replied:
Done.

rniwa commented on Source/WebKit/WebProcess/InjectedBundle/DOM/InjectedBundleRangeHandle.cpp:
```diff
@@ -175,7 +175,8 @@ RefPtr<WebImage> InjectedBundleRangeHandle::renderedImage(SnapshotOptions option
 {
     auto range = makeSimpleRange(m_range);
-    range.start.document().updateLayout();
+    Ref document = range.start.document();
```
Why not `range.start.protectedDocument()->updateLayout()`?

rwlbuis replied:
Done.

rwlbuis committed:
Adopt smart pointers for updateLayout calls
https://bugs.webkit.org/show_bug.cgi?id=272738

Reviewed by Ryosuke Niwa.

Adopt smart pointers for updateLayout calls.

* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::propertyValue const):
* Source/WebCore/dom/BoundaryPoint.h:
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::firstRectForRange const):
* Source/WebCore/html/HTMLLabelElement.cpp:
(WebCore::HTMLLabelElement::focus):
* Source/WebCore/html/ImageInputType.cpp:
(WebCore::ImageInputType::height const):
(WebCore::ImageInputType::width const):
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.cpp:
(WebCore::MediaControlTextTrackContainerElement::createTextTrackRepresentationImage):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::performSizeToContentAutoSize):
(WebCore::LocalFrameView::setTracksRepaints):
(WebCore::LocalFrameView::trackedRepaintRectsAsText const):
* Source/WebCore/page/PrintContext.cpp:
(WebCore::PrintContext::pageProperty):
* Source/WebCore/page/ios/FrameIOS.mm:
(WebCore::LocalFrame::wordsInCurrentParagraph const):
(WebCore::LocalFrame::preferredHeight const):
(WebCore::LocalFrame::updateLayout const):
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
(WebKit::WebFullScreenManager::willEnterFullScreen):
* Source/WebKit/WebProcess/InjectedBundle/DOM/InjectedBundleRangeHandle.cpp:
(WebKit::InjectedBundleRangeHandle::text const):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::updateAttachmentAttributes):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::requestDocumentEditingContext):
(WebKit::WebPage::focusTextInputContextAndPlaceCaret):

Canonical link: https://commits.webkit.org/277606@main
